neo/SmartContract/NeoService.cs

```diff
@@ -173,13 +174,24 @@ private bool Blockchain_GetAccount(ExecutionEngine engine)
             return true;
         }
 
-        private bool Blockchain_GetValidators(ExecutionEngine engine)
+        internal bool Blockchain_GetValidators(ExecutionEngine engine)
         {
             ECPoint[] validators = Snapshot.GetValidators();
             engine.CurrentContext.EvaluationStack.Push(validators.Select(p => (StackItem)p.EncodePoint(true)).ToArray());
             return true;
         }
 
+        internal bool Blockchain_GetValidatorsScript(ExecutionEngine engine)
+        {
+            // Note: this must be compatible with Consensus validation service
+            // In the future, we can put this in some core part shared among both components (NeoService/ConsensusService)
+            ECPoint[] validators = Snapshot.GetValidators();
+            int m = validators.Length - (validators.Length - 1) / 3;
+            Contract contract = Contract.CreateMultiSigContract(m, validators);
```

Comment:
Maybe we can try to cache this contract?

Reply:
It would be very nice. Where can we put this?
Although this is used here and on Consensus service, it looks to me that this is a resposibility of the Ledger/Blockchain part. But if we do that, it will be innaccessible to NeoService (unless it has access to NeoSystem, which is also possible).

Reply:
Good idea, Shargon, it will surely avoid extra calculation if contracts start to use this script.